bundles/com.espressif.idf.core/src/com/espressif/idf/core/toolchain/IDFTargets.java

```diff
@@ -0,0 +1,270 @@
+	private List<IDFTarget> previewTargets;
+
+	public IDFTargets()
+	{
+		this.supportedTargets = new ArrayList<>();
+		this.previewTargets = new ArrayList<>();
+	}
+
+	public void addSupportedTarget(String target)
+	{
+		supportedTargets.add(new IDFTarget(target, false));
+	}
+
+	public void addPreviewTarget(String target)
+	{
+		previewTargets.add(new IDFTarget(target, true));
+	}
+
+	public List<IDFTarget> getAllTargets()
+	{
+		List<IDFTarget> allTargets = new ArrayList<>();
+		allTargets.addAll(supportedTargets);
+		allTargets.addAll(previewTargets);
+		return allTargets;
+	}
+
+	public List<IDFTarget> getSupportedTargets()
+	{
+		return supportedTargets;
+	}
+
+	public List<IDFTarget> getPreviewTargets()
+	{
+		return previewTargets;
+	}
+
+	public boolean hasTarget(String targetName)
+	{
+		return getAllTargets().stream().anyMatch(target -> target.getName().equals(targetName));
+	}
+
+	/**
+	 * Get a specific target by name
+	 * 
+	 * @param targetName Name of the target to find
+	 * @return IDFTarget if found, null otherwise
+	 */
+	public IDFTarget getTarget(String targetName)
+	{
+		return getAllTargets().stream().filter(target -> target.getName().equals(targetName)).findFirst().orElse(null);
+	}
+
+	/**
+	 * Get all target names as strings
+	 * 
+	 * @return List of target names
+	 */
+	public List<String> getAllTargetNames()
+	{
+		return getAllTargets().stream().map(IDFTarget::getName).collect(java.util.stream.Collectors.toList());
+	}
+
+	/**
+	 * Get supported target names as strings
+	 * 
+	 * @return List of supported target names
+	 */
+	public List<String> getSupportedTargetNames()
+	{
+		return getSupportedTargets().stream().map(IDFTarget::getName).collect(java.util.stream.Collectors.toList());
+	}
+
+	/**
+	 * Get preview target names as strings
+	 * 
+	 * @return List of preview target names
+	 */
+	public List<String> getPreviewTargetNames()
+	{
+		return getPreviewTargets().stream().map(IDFTarget::getName).collect(java.util.stream.Collectors.toList());
+	}
+
+	/**
+	 * Inner class representing a single IDF target
+	 */
+	public static class IDFTarget
+	{
+		private final String name;
+		private final boolean isPreview;
+
+		public IDFTarget(String name, boolean isPreview)
+		{
+			this.name = name;
+			this.isPreview = isPreview;
+		}
+
+		public String getName()
+		{
+			return name;
+		}
+
+		public boolean isPreview()
+		{
+			return isPreview;
+		}
+
+		/**
+		 * Get the architecture for this target
+		 * 
+		 * @return "xtensa" for esp32/esp32s2/esp32s3, "riscv32" for others
+		 */
+		public String getArchitecture()
+		{
+			switch (name)
+			{
+			case "esp32": //$NON-NLS-1$
+			case "esp32s2": //$NON-NLS-1$
+			case "esp32s3": //$NON-NLS-1$
+				return "xtensa"; //$NON-NLS-1$
+			case "esp32c2": //$NON-NLS-1$
+			case "esp32c3": //$NON-NLS-1$
+			case "esp32c6": //$NON-NLS-1$
+			case "esp32h2": //$NON-NLS-1$
+			case "esp32p4": //$NON-NLS-1$
+			default:
+				return "riscv32"; //$NON-NLS-1$
+			}
+		}
```

Comment:
NIT:
We could simplify the getArchitecture() method to make it more readable and avoid multiple riscv32 statements or returns. For example, using Java 21 switch expressions:
```
public String getArchitecture() {
    return switch (name) {
        case "esp32", "esp32s2", "esp32s3" -> "xtensa";
        default -> "riscv32";
    };
}
```

Alternatively, a Set-based approach could be used
```
private static final Set<String> XTENSA_CHIPS = Set.of("esp32", "esp32s2", "esp32s3");

public String getArchitecture() {
    return XTENSA_CHIPS.contains(name) ? "xtensa" : "riscv32";
}
```